### PR TITLE
Functional rewrite

### PR DESCRIPTION
**Pull request by AHsu98**
Overall rewrite to clean up and make it more functional. Before, everything was a bit overcomplicated and wrapped up in classes. Generally trying to just pass things around here rather than making them class attributes.

### CODE REVIEW

File: src/pydisagg/DisaggModel.py [hunk 1]
```diff
@@ -21,574 +21,677 @@ class DisaggModel:
...
-            Inflation factor generated by Delta method., by default None
-        beta_standard_error : Optional[float], optional
-            Standard error of beta parameter, by default None
         """
 
-        self.beta_parameter = beta_parameter
-        self.beta_standard_error = beta_standard_error
-        self.error_inflation = error_inflation
-
         self.parameter_transformation = parameter_transformation
```
Comment (kels271828) [inline src/pydisagg/DisaggModel.py:36]: Why do you need both self.parameter_transformation and self.T?

File: src/pydisagg/DisaggModel.py [hunk 2]
```diff
@@ -21,574 +21,677 @@ class DisaggModel:
...
-        observed_total_se : Optional[float], optional
-            se of observed_total, by default None
-        rate_pattern : Optional[NDArray], optional
-            , by default None
+            aggregated observation
+        rate_pattern : NDArray
+            rate pattern to assume (generalized) proportionality to
+        bucket_populations : NDArray
+            populations in each bucket
         lower_guess : Optional[float], optional
```
Comment (kels271828) [inline src/pydisagg/DisaggModel.py:156]: I would use `lower_guess : float, optional` instead

File: src/pydisagg/models.py [hunk 3]
```diff
@@ -18,53 +18,25 @@ class RateMultiplicativeModel(DisaggModel):
```
Comment (kels271828) [inline src/pydisagg/models.py:18]: I noticed in transformations.py, you have the abstract class in the same module as the other classes. Why did you choose to have DisaggModel in a different module than your other model classes?

**Review by kels271828 — COMMENTED** [review]
Since I'm less familiar with the methods, my comments are mostly about structure and formatting. Your code seems really modular which is great. For docstrings, you could go through and make sure that your formatting is consistent (spacing, punctuation, capitalization, etc.). I would also add class attributes to the class docstrings. For code, you could use a linter to point out inconsistencies or issues (e.g., pylint or flake8), or try the black package to automatically reformat things. David suggested separating feature commits from formatting commits to make it easier to review, and you could also wait to fix formatting until the very end when you know your code is working.

File: src/pydisagg/models.py (hunk 3 above)
Comment (AHsu98) [reply src/pydisagg/models.py:18]: Both of these choices were somewhat arbitrary, I'll split them both up for consistency?

Comment (AHsu98) [reply src/pydisagg/models.py:18]: Fixed in the commit below.

**Comment by AHsu98** [comment]
@kels271828 I agree with the idea of separating features from refactoring, but I got confused when I was trying to fix/add features first (before I realized I should try to refactor), which is where this whole rewrite idea came from. I'll add some more comments after I finish this on what exactly has changed, and explaining feature vs refactoring differences.

**Comment by AHsu98** [comment]
Oh I just remembered I missed some linting type stuff, and was going to make another commit from what @kels271828 had mentioned above, but besides that, we are all set here.

File: src/pydisagg/DisaggModel.py (hunk 1 above)
Comment (AHsu98) [reply src/pydisagg/DisaggModel.py]: fixed in [commit here](commit ee0ba9e3a212f0fd7abb2bd0f77607a7e3137500)

File: src/pydisagg/DisaggModel.py [hunk 4]
```diff
@@ -21,574 +21,677 @@ class DisaggModel:
     We do this because it works better with the delta method calculations, and we assume that we
     have a log in the transormations so additive factors become multiplicative
     '''
```
[the remark this answers is not in the document]
Comment (AHsu98) [reply src/pydisagg/DisaggModel.py]: Fixed in commit e2317bc98bacda4b838f6fd6cf74fe5e2ce51868

File: src/pydisagg/DisaggModel.py (hunk 2 above)
Comment (AHsu98) [reply src/pydisagg/DisaggModel.py]: Thanks for pointing this out, fixed in commit acb2254ae7d4ef4fdcb10662c6dcd656110c35ce

**Comment by AHsu98** [comment]
Alright, I think we should be all set here! @kels271828 @zhengp0 @saravkin, I should be ready to merge now.

**Comment by AHsu98** [comment]
Had a quick chat with sasha, going to go ahead and just force merge this now.

**Pull request merged**